gno.land/pkg/gnoland/types.go

```diff
@@ -102,6 +221,200 @@ func ProtoGnoAccount() std.Account {
 	return &GnoAccount{}
 }
 
+// GnoSession extends BaseSession with ACL capabilities
+//
+// Currently, a session is linked to a specific public key, which means authentication
+// and authorization are tied to a particular cryptographic identity. This design allows
+// for straightforward signature-based verification.
+//
+// In future iterations, the session concept could be extended beyond just public keys.
+// For example, sessions might be:
+// - Linked to smart contracts that implement custom authorization logic
+// - Associated with multi-sig requirements
+// - Connected to external identity providers
+// - Managed by governance protocols
+//
+// This would allow for more sophisticated authentication and delegation mechanisms
+// while keeping the core session abstraction intact.
+type GnoSession struct {
+	std.BaseSession
+	// Access Control Lists using BitSet
+	Flags                 BitSet    `json:"flags" yaml:"flags"`
+	ExpirationTime        time.Time `json:"expiration_time" yaml:"expiration_time"`
+	CoinsTransferCapacity std.Coins `json:"coins_transfer_capacity" yaml:"coins_transfer_capacity"`
```

Comment:
is this like a credit line? 
how about throttled, but even with a session if it has no deadline/expiry, allow it to fully drain an account over say 10 years? This can come in handy for recovery.